StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/UpdatePaymentMethodViewController.swift

```diff
@@ -278,11 +278,31 @@ final class UpdatePaymentMethodViewController: UIViewController {
         updateButton.update(state: updateParams != nil || hasChangedDefaultPaymentMethodCheckbox ? .enabled : .disabled)
     }
 
-    func hasChangedFields(original: STPPaymentMethodCard, updated: STPPaymentMethodCardParams) -> Bool {
-        let cardBrandChanged = configuration.canUpdateCardBrand && original.preferredDisplayBrand != updated.networks?.preferred?.toCardBrand
-        let updatedMM = NSNumber(value: original.expMonth) != updated.expMonth
-        let updatedYY = original.twoDigitYear != updated.expYear
-        return cardBrandChanged || updatedMM || updatedYY
+    func hasChangedCardOrBillingAddressFields(originalPaymentMethod: STPPaymentMethod, updatedPaymentMethodParams: STPPaymentMethodParams) -> Bool {
+
+        guard originalPaymentMethod.type == .card,
+              let originalCardPaymentMethod = originalPaymentMethod.card else {
+            return false
+        }
```

Comment:
I guess we're only intending for cards to be updated in this PR, but are we expecting to be able to update SEPA or others? If so I wonder if there is a better way to model this from the start so it's easy to add in new PMs. I'm assuming for SEPA and others we'd only care about billingParamsChanged and not cardBrandChanged or cardParamsChanged?

Should we assert here if it's not a card PM?

Reply:
For this project we will ONLY support updating cards. Updating other payment methods is not in scope, and not likely to happen in the near future.

Tactically, yes, lets stpAssert! great idea!